www/include/monitoring/status/Services/xml/serviceXML.php

```diff
@@ -84,85 +70,80 @@
-$hostgroups = $obj->checkArgument("hostgroups", $_GET, $obj->defaultHostgroups);
-$servicegroups = $obj->checkArgument("servicegroups", $_GET, $obj->defaultServicegroups);
-$search = $obj->checkArgument("search", $_GET, "");
-$search_host = $obj->checkArgument("search_host", $_GET, "");
-$search_output = $obj->checkArgument("search_output", $_GET, "");
-$sort_type = $obj->checkArgument("sort_type", $_GET, "host_name");
-$order = $obj->checkArgument("order", $_GET, "ASC");
-$dateFormat = $obj->checkArgument("date_time_format_status", $_GET, "Y/m/d H:i:s");
-$search_type_host = $obj->checkArgument("search_type_host", $_GET, 1);
-$search_type_service = $obj->checkArgument("search_type_service", $_GET, 1);
-$criticality_id = $obj->checkArgument('criticality', $_GET, $obj->defaultCriticality);
-
-$statusService = $obj->checkArgument("statusService", $_GET, "");
-$statusFilter = $obj->checkArgument("statusFilter", $_GET, "");
-
-CentreonDb::checkInjection($o);
-CentreonDb::checkInjection($p);
-CentreonDb::checkInjection($nc);
-CentreonDb::checkInjection($num);
-CentreonDb::checkInjection($limit);
-CentreonDb::checkInjection($instance);
-CentreonDb::checkInjection($hostgroups);
-CentreonDb::checkInjection($servicegroups);
-CentreonDb::checkInjection($search);
-CentreonDb::checkInjection($search_host);
-CentreonDb::checkInjection($search_output);
-CentreonDb::checkInjection($sort_type);
-CentreonDb::checkInjection($order);
-CentreonDb::checkInjection($dateFormat);
-CentreonDb::checkInjection($search_type_host);
-CentreonDb::checkInjection($search_type_service);
-CentreonDb::checkInjection($criticality_id);
-
-/* Store in session the last type of call */
+// Check Arguments From GET tab
+$o = filter_input(INPUT_GET, 'o', FILTER_SANITIZE_STRING, array('options' => array('default' => 'h')));
+$p = filter_input(INPUT_GET, 'p', FILTER_VALIDATE_INT, array('options' => array('default' => 2)));
+$num = filter_input(INPUT_GET, 'num', FILTER_VALIDATE_INT, array('options' => array('default' => 0)));
+$limit = filter_input(INPUT_GET, 'limit', FILTER_VALIDATE_INT, array('options' => array('default' => 20)));
+$nc = filter_input(INPUT_GET, 'criticality', FILTER_VALIDATE_INT, array('options' => array('default' => 0)));
+$criticality_id = filter_input(INPUT_GET, 'nc', FILTER_VALIDATE_INT, array('options' => array('default' => 0)));
+$search = filter_input(INPUT_GET, 'search', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+$search_host = filter_input(
+    INPUT_GET,
+    'search_host',
+    FILTER_SANITIZE_STRING,
+    array('options' => array('default' => ''))
+);
+$search_output = filter_input(
+    INPUT_GET,
+    'search_output',
+    FILTER_SANITIZE_STRING,
+    array('options' => array('default' => ''))
+);
+$sort_type = filter_input(
+    INPUT_GET,
+    'sort_type',
+    FILTER_SANITIZE_STRING,
+    array('options' => array('default' => 'host_name'))
+);
+$order = filter_input(
```

Comment:
Please stop using regexes to simply test a value with two possible cases.
Do not use a mass to drive a nail

Reply:
I'll replace these by whitelist ASAP ;)